CONTRIBUTING.md

```diff
@@ -0,0 +1,111 @@
+and [Gubernator](https://github.com/kubernetes/test-infra/tree/master/gubernator)
+(see [Knative Prow](https://github.com/knative/test-infra/blob/master/ci/prow/prow_setup.md)).
+This means that automation will be applied to your pull requests.
+
+Prow has a [number of commands](https://prow.knative.dev/command-help) you can use to interact with it,
+in particular:
+
+* Before a PR can be merged, an owner must approve it with both `/lgtm` AND `/approve`
+* If you don't a PR to be merged, e.g. so that the author can make changes, you can put it on hold with `/hold`
+* When tests (run by Prow) fail, it will add a comment to the PR with the commands to re-run any failing tests
+* You can add dog and cat pictures with `/woof` and `/meow`
+
+Any changes will cause the `/lgtm` label to be removed and it will need to be re-applied.
+
+_See also [Knative docs on reviewing](https://github.com/knative/docs/blob/master/community/REVIEWING.md)._
+
+## Commit Messagse
+
+All commit messages should follow [these best practices](https://chris.beams.io/posts/git-commit/),
+specifically:
+
+* Start with a subject line
+* Contain a body that explains _why_ you're making the change you're making
+
+Aim for [2 paragraphs in the body](https://www.youtube.com/watch?v=PJjmw9TRB7s).
+Not sure what to put? Include:
+
+* What is the problem being solved?
+* Why is this the best approach?
+* What other approaches did you consider?
+* What side effects will this approach have?
+* What future work remains to be done?
+
+## Roadmap and contributions
+
+As of Sept 2018, our roadmap for the next few months is to:
+
+1. Soldify the API, including:
+
+   * [Designing conditional execution](https://github.com/knative/build-pipeline/issues/27)
+   * [Designing sources](https://github.com/knative/build-pipeline/issues/13)
+   * [Ensuring the the same inputs/outputs are used throughout the pipeline](https://github.com/knative/build-pipeline/issues/11)
+   * [Design references between objects](https://github.com/knative/build-pipeline/issues/38)
+   * [Design templating](https://github.com/knative/build-pipeline/issues/36)
+   * [Design results](https://github.com/knative/build-pipeline/issues/37)
+
+2. Complete a user study (see issues labeled with [user-study](https://github.com/knative/build-pipeline/issues?q=is%3Aissue+is%3Aopen+label%3Auser-study))
+
+3. [Create an initial POC command line tool for interacting with Pipelines](https://github.com/knative/build-pipeline/issues/35)
+
+### Contributions wanted
+
+* To find issues that we particularly would like contributors to tackle, look for
+  [issues with the "help wanted" label](https://github.com/knative/build-pipeline/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22).
+* Issues that are good for new folks will additionally be marked with
+  ["good first issue"](https://github.com/knative/build-pipeline/issues?q=is%3Aissue+is%3Aopen+label%3A%22good+first+issue%22).
```

Comment:
feedback from @mattmoor : we should use labels like `kind/good-first-issue` instead which can be applied by prow  _whispers: even though that deviates from [github standards](https://help.github.com/articles/helping-new-contributors-find-your-project-with-labels/)_

Reply:
hey actually @mattmoor i'm going to hold off on this for now, I don't want to miss out on any automation/etc. github has around the standard labels, but ill change this if i can be convinced!

i feel like these labels should be applied only by maintainers of the project anyway while triaging